vignettes/column-types.Rmd

```diff
@@ -14,70 +14,29 @@ knitr::opts_chunk$set(
-Due to its different design, vroom is able to sub-sample `n` rows throughout the file and it always includes the last row, whereas earlier versions of readr just consulted the first `n` rows.
-In practice, the result is that the default of `guess_max = min(1000, n_max)` produces better guessed column types that it used to.
-It should feel less necessary to fiddle with `guess_max` now.
-
-<!-- 
-https://github.com/r-lib/vroom/issues/352
+## Automatic guessing
 
-In fact vroom generally does this guessing better than readr currently does. Readr always uses the first guess_max number of rows for guessing, whereas vroom uses rows interspersed throughout the file if guess_max is less than the total number of rows.
+If you don't explicit specify column types with the `col_types` argument readr will attempt to guess them using some simple heuristics.
+By default, it will inspect the values of 1000 evenly spaced values from the first to the last row.
+This is a heuristic designed to always be fast (no matter how large your file is) and, in our experience, does a good job in most cases.
 
-However as this example shows we should probably always include the last row when guessing, so I have made a change to always include the last row in the guess in the future.
--->
+If needed, you can request that readr uses more rows by supplying the `guess_max` argument.
+You can even supply `guess_max = Inf` to use every row to guess the column types.
+You might wonder why this isn't the default.
+That's because it's slow: it has to look at every column twice, once to determine the type and once to parse the value.
+In most cases you're best off supplying `col_types` yourself.
 
-As always, remember that the best strategy is to provide explicit column types as any data analysis project matures past the exploratory phase.
+### Legacy behavior
 
-<!-- Future link to a vignette on column specification or, if that content gets co-located here, link to later in this vignette. -->
-
-## readr first edition and readr < 2.0.0
-
-The parsing engine in readr versions prior to 2.0.0 is now called the first edition.
-If you're using readr >= 2.0.0, you can still access first edition parsing via the functions `with_edition()` and `local_edition()`.
-And, obviously, if you're using readr < 2.0.0, you will get first edition parsing, by definition, because that's all there is.
-
-The first edition parser doesn't have a perfect way to convey "use all of the data to guess the column types".
-(This is one of several reasons to prefer readr >= 2.0.0.)
-
-Let's set up a slightly tricky file, so we can demonstrate different approaches.
-The column `x` is mostly empty, but has some numeric data at the very end, in row 1001.
+Column guessing was substantially worse in the first edition because it always looked at the first 1000 rows, and through some application of Murphy's law it appears that many real csv files have different structure in the final rows.
```

Comment:
```suggestion
Column type guessing was substantially worse in the first edition of readr (meaning, prior to v2.0.0), because it always looked at the first 1000 rows, and through some application of Murphy's Law, it appears that many real csv files have lots of empty values at the start, followed by more "excitement" later in the file.
```

This wording is pretty informal, but seems compatible with invoking Murphy's Law.